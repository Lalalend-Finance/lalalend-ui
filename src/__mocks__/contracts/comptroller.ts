import { Comptroller } from 'types/contracts_evmos';

const comptrollerResponses: {
  miaSEBVaultRate: Awaited<
    ReturnType<ReturnType<Comptroller['methods']['miaSEBVaultRate']>['call']>
  >;
  getHypotheticalAccountLiquidity: Awaited<
    ReturnType<ReturnType<Comptroller['methods']['getHypotheticalAccountLiquidity']>['call']>
  >;
} = {
  miaSEBVaultRate: '5000000000',
  getHypotheticalAccountLiquidity: {
    0: '100000000',
    1: '200000000',
    2: '300000000',
  },
};

export default comptrollerResponses;
