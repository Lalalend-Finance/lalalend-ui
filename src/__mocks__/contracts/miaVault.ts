import { MiaVault } from 'types/contracts_evmos';

const miaVaultResponses: {
  userInfo: Awaited<ReturnType<ReturnType<MiaVault['methods']['getUserInfo']>['call']>>;
  poolInfo: Awaited<ReturnType<ReturnType<MiaVault['methods']['poolInfos']>['call']>>;
  poolLength: Awaited<ReturnType<ReturnType<MiaVault['methods']['poolLength']>['call']>>;
  totalAllocPoints: Awaited<
    ReturnType<ReturnType<MiaVault['methods']['totalAllocPoints']>['call']>
  >;
  rewardTokenAmountsPerBlock: Awaited<
    ReturnType<ReturnType<MiaVault['methods']['rewardTokenAmountsPerBlock']>['call']>
  >;
  pendingReward: Awaited<ReturnType<ReturnType<MiaVault['methods']['pendingReward']>['call']>>;
  getWithdrawalRequests: Awaited<
    ReturnType<ReturnType<MiaVault['methods']['getWithdrawalRequests']>['call']>
  >;
} = {
  userInfo: {
    pendingWithdrawals: '1000000000000000000',
    rewardDebt: '2000000000000000000',
    amount: '30000000000000000000',
    0: '1000000000000000000',
    1: '2000000000000000000',
    2: '30000000000000000000',
  },
  poolInfo: {
    token: '0x4B7268FC7C727B88c5Fc127D41b491BfAe63e144',
    allocPoint: '10',
    lastRewardBlock: '100000',
    accRewardPerShare: '123871680',
    lockPeriod: '200',
    0: '0x4B7268FC7C727B88c5Fc127D41b491BfAe63e144',
    1: '10',
    2: '100000',
    3: '123871680',
    4: '200',
  },
  poolLength: '5',
  totalAllocPoints: '100',
  rewardTokenAmountsPerBlock: '10000000',
  pendingReward: '200000000',
  getWithdrawalRequests: [
    ['1000000000000000000', '1656499404'],
    ['2000000000000000000', '1656599404'],
    ['3000000000000000000', '1656699404'],
  ],
};

export default miaVaultResponses;
