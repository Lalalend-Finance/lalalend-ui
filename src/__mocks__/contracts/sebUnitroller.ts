import { SebUnitroller } from 'types/contracts_evmos';

const sebUnitrollerResponses: {
  getMintableSEB: Awaited<
    ReturnType<ReturnType<SebUnitroller['methods']['getMintableSEB']>['call']>
  >; 
} = {
  getMintableSEB: {
    0: '20000000000000000000',
    1: '40000000000000000000',
  },
};

export default sebUnitrollerResponses;
