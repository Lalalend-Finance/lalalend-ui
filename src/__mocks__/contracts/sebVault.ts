import { SebVault } from 'types/contracts_evmos';

const sebVaultResponses: {
  userInfo: Awaited<ReturnType<ReturnType<SebVault['methods']['userInfo']>['call']>>;
  pendingMIA: Awaited<ReturnType<ReturnType<SebVault['methods']['pendingMIA']>['call']>>;
} = {
  userInfo: {
    amount: '100000000000000000000000',
    rewardDebt: '2000',
    0: '100000000000000000000000',
    1: '4000',
  },
  pendingMIA: '600000000',
};

export default sebVaultResponses;
