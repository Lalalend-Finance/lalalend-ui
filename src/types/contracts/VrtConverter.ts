/* Autogenerated file. Do not edit manually. */

/* tslint:disable */

/* eslint-disable */
import type {
  Callback,
  PayableTransactionObject,
  NonPayableTransactionObject,
  BlockType,
  ContractEventLog,
  BaseContract,
} from "./types";
import type BN from "bn.js";
import type { EventEmitter } from "events";
import type { EventLog } from "web3-core";
import type { ContractOptions } from "web3-eth-contract";

export interface EventOptions {
  filter?: object;
  fromBlock?: BlockType;
  topics?: string[];
}

export type ConversionInfoSet = ContractEventLog<{
  conversionRatio: string;
  conversionStartTime: string;
  conversionPeriod: string;
  conversionEndTime: string;
  0: string;
  1: string;
  2: string;
  3: string;
}>;
export type NewAdmin = ContractEventLog<{
  oldAdmin: string;
  newAdmin: string;
  0: string;
  1: string;
}>;
export type NewPendingAdmin = ContractEventLog<{
  oldPendingAdmin: string;
  newPendingAdmin: string;
  0: string;
  1: string;
}>;
export type TokenConverted = ContractEventLog<{
  reedeemer: string;
  vrtAddress: string;
  vrtAmount: string;
  miaAddress: string;
  miaAmount: string;
  0: string;
  1: string;
  2: string;
  3: string;
  4: string;
}>;
export type TokenWithdraw = ContractEventLog<{
  token: string;
  to: string;
  amount: string;
  0: string;
  1: string;
  2: string;
}>;
export type MIAVestingSet = ContractEventLog<{
  miaVestingAddress: string;
  0: string;
}>;

export interface VrtConverter extends BaseContract {
  constructor(
    jsonInterface: any[],
    address?: string,
    options?: ContractOptions
  ): VrtConverter;
  clone(): VrtConverter;
  methods: {
    DEAD_ADDRESS(): NonPayableTransactionObject<string>;

    _acceptAdmin(): NonPayableTransactionObject<void>;

    _setPendingAdmin(
      newPendingAdmin: string
    ): NonPayableTransactionObject<void>;

    _setMIAVesting(
      _miaVestingAddress: string
    ): NonPayableTransactionObject<void>;

    admin(): NonPayableTransactionObject<string>;

    conversionEndTime(): NonPayableTransactionObject<string>;

    conversionPeriod(): NonPayableTransactionObject<string>;

    conversionRatio(): NonPayableTransactionObject<string>;

    conversionStartTime(): NonPayableTransactionObject<string>;

    convert(vrtAmount: number | string | BN): NonPayableTransactionObject<void>;

    isConversionActive(): NonPayableTransactionObject<boolean>;

    lastDayUpdated(): NonPayableTransactionObject<string>;

    pendingAdmin(): NonPayableTransactionObject<string>;

    totalVrtConverted(): NonPayableTransactionObject<string>;

    vrt(): NonPayableTransactionObject<string>;

    vrtDecimalsMultiplier(): NonPayableTransactionObject<string>;

    mia(): NonPayableTransactionObject<string>;

    miaDecimalsMultiplier(): NonPayableTransactionObject<string>;

    miaVesting(): NonPayableTransactionObject<string>;
  };
  events: {
    ConversionInfoSet(cb?: Callback<ConversionInfoSet>): EventEmitter;
    ConversionInfoSet(
      options?: EventOptions,
      cb?: Callback<ConversionInfoSet>
    ): EventEmitter;

    NewAdmin(cb?: Callback<NewAdmin>): EventEmitter;
    NewAdmin(options?: EventOptions, cb?: Callback<NewAdmin>): EventEmitter;

    NewPendingAdmin(cb?: Callback<NewPendingAdmin>): EventEmitter;
    NewPendingAdmin(
      options?: EventOptions,
      cb?: Callback<NewPendingAdmin>
    ): EventEmitter;

    TokenConverted(cb?: Callback<TokenConverted>): EventEmitter;
    TokenConverted(
      options?: EventOptions,
      cb?: Callback<TokenConverted>
    ): EventEmitter;

    TokenWithdraw(cb?: Callback<TokenWithdraw>): EventEmitter;
    TokenWithdraw(
      options?: EventOptions,
      cb?: Callback<TokenWithdraw>
    ): EventEmitter;

    MIAVestingSet(cb?: Callback<MIAVestingSet>): EventEmitter;
    MIAVestingSet(
      options?: EventOptions,
      cb?: Callback<MIAVestingSet>
    ): EventEmitter;

    allEvents(options?: EventOptions, cb?: Callback<EventLog>): EventEmitter;
  };

  once(event: "ConversionInfoSet", cb: Callback<ConversionInfoSet>): void;
  once(
    event: "ConversionInfoSet",
    options: EventOptions,
    cb: Callback<ConversionInfoSet>
  ): void;

  once(event: "NewAdmin", cb: Callback<NewAdmin>): void;
  once(event: "NewAdmin", options: EventOptions, cb: Callback<NewAdmin>): void;

  once(event: "NewPendingAdmin", cb: Callback<NewPendingAdmin>): void;
  once(
    event: "NewPendingAdmin",
    options: EventOptions,
    cb: Callback<NewPendingAdmin>
  ): void;

  once(event: "TokenConverted", cb: Callback<TokenConverted>): void;
  once(
    event: "TokenConverted",
    options: EventOptions,
    cb: Callback<TokenConverted>
  ): void;

  once(event: "TokenWithdraw", cb: Callback<TokenWithdraw>): void;
  once(
    event: "TokenWithdraw",
    options: EventOptions,
    cb: Callback<TokenWithdraw>
  ): void;

  once(event: "MIAVestingSet", cb: Callback<MIAVestingSet>): void;
  once(
    event: "MIAVestingSet",
    options: EventOptions,
    cb: Callback<MIAVestingSet>
  ): void;
}
