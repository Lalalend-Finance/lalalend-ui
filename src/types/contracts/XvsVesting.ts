/* Autogenerated file. Do not edit manually. */

/* tslint:disable */

/* eslint-disable */
import type {
  Callback,
  PayableTransactionObject,
  NonPayableTransactionObject,
  BlockType,
  ContractEventLog,
  BaseContract,
} from "./types";
import type BN from "bn.js";
import type { EventEmitter } from "events";
import type { EventLog } from "web3-core";
import type { ContractOptions } from "web3-eth-contract";

export interface EventOptions {
  filter?: object;
  fromBlock?: BlockType;
  topics?: string[];
}

export type NewAdmin = ContractEventLog<{
  oldAdmin: string;
  newAdmin: string;
  0: string;
  1: string;
}>;
export type NewPendingAdmin = ContractEventLog<{
  oldPendingAdmin: string;
  newPendingAdmin: string;
  0: string;
  1: string;
}>;
export type VRTConversionSet = ContractEventLog<{
  vrtConversionAddress: string;
  0: string;
}>;
export type VestedTokensClaimed = ContractEventLog<{
  recipient: string;
  amountClaimed: string;
  0: string;
  1: string;
}>;
export type MIAVested = ContractEventLog<{
  recipient: string;
  startTime: string;
  amount: string;
  withdrawnAmount: string;
  0: string;
  1: string;
  2: string;
  3: string;
}>;
export type MIAWithdrawn = ContractEventLog<{
  recipient: string;
  amount: string;
  0: string;
  1: string;
}>;

export interface XvsVesting extends BaseContract {
  constructor(
    jsonInterface: any[],
    address?: string,
    options?: ContractOptions
  ): XvsVesting;
  clone(): XvsVesting;
  methods: {
    _acceptAdmin(): NonPayableTransactionObject<void>;

    _setPendingAdmin(
      newPendingAdmin: string
    ): NonPayableTransactionObject<void>;

    _setVRTConversion(
      _vrtConversionAddress: string
    ): NonPayableTransactionObject<void>;

    admin(): NonPayableTransactionObject<string>;

    deposit(
      recipient: string,
      depositAmount: number | string | BN
    ): NonPayableTransactionObject<void>;

    getCurrentTime(): NonPayableTransactionObject<string>;

    getVestedAmount(recipient: string): NonPayableTransactionObject<string>;

    getWithdrawableAmount(recipient: string): NonPayableTransactionObject<{
      totalWithdrawableAmount: string;
      totalVestedAmount: string;
      totalWithdrawnAmount: string;
      0: string;
      1: string;
      2: string;
    }>;

    numberOfVestings(arg0: string): NonPayableTransactionObject<string>;

    pendingAdmin(): NonPayableTransactionObject<string>;

    vestings(
      arg0: string,
      arg1: number | string | BN
    ): NonPayableTransactionObject<{
      recipient: string;
      startTime: string;
      amount: string;
      withdrawnAmount: string;
      0: string;
      1: string;
      2: string;
      3: string;
    }>;

    vrtConversionAddress(): NonPayableTransactionObject<string>;

    withdraw(): NonPayableTransactionObject<void>;

    mia(): NonPayableTransactionObject<string>;

    miaDecimalsMultiplier(): NonPayableTransactionObject<string>;
  };
  events: {
    NewAdmin(cb?: Callback<NewAdmin>): EventEmitter;
    NewAdmin(options?: EventOptions, cb?: Callback<NewAdmin>): EventEmitter;

    NewPendingAdmin(cb?: Callback<NewPendingAdmin>): EventEmitter;
    NewPendingAdmin(
      options?: EventOptions,
      cb?: Callback<NewPendingAdmin>
    ): EventEmitter;

    VRTConversionSet(cb?: Callback<VRTConversionSet>): EventEmitter;
    VRTConversionSet(
      options?: EventOptions,
      cb?: Callback<VRTConversionSet>
    ): EventEmitter;

    VestedTokensClaimed(cb?: Callback<VestedTokensClaimed>): EventEmitter;
    VestedTokensClaimed(
      options?: EventOptions,
      cb?: Callback<VestedTokensClaimed>
    ): EventEmitter;

    MIAVested(cb?: Callback<MIAVested>): EventEmitter;
    MIAVested(options?: EventOptions, cb?: Callback<MIAVested>): EventEmitter;

    MIAWithdrawn(cb?: Callback<MIAWithdrawn>): EventEmitter;
    MIAWithdrawn(
      options?: EventOptions,
      cb?: Callback<MIAWithdrawn>
    ): EventEmitter;

    allEvents(options?: EventOptions, cb?: Callback<EventLog>): EventEmitter;
  };

  once(event: "NewAdmin", cb: Callback<NewAdmin>): void;
  once(event: "NewAdmin", options: EventOptions, cb: Callback<NewAdmin>): void;

  once(event: "NewPendingAdmin", cb: Callback<NewPendingAdmin>): void;
  once(
    event: "NewPendingAdmin",
    options: EventOptions,
    cb: Callback<NewPendingAdmin>
  ): void;

  once(event: "VRTConversionSet", cb: Callback<VRTConversionSet>): void;
  once(
    event: "VRTConversionSet",
    options: EventOptions,
    cb: Callback<VRTConversionSet>
  ): void;

  once(event: "VestedTokensClaimed", cb: Callback<VestedTokensClaimed>): void;
  once(
    event: "VestedTokensClaimed",
    options: EventOptions,
    cb: Callback<VestedTokensClaimed>
  ): void;

  once(event: "MIAVested", cb: Callback<MIAVested>): void;
  once(
    event: "MIAVested",
    options: EventOptions,
    cb: Callback<MIAVested>
  ): void;

  once(event: "MIAWithdrawn", cb: Callback<MIAWithdrawn>): void;
  once(
    event: "MIAWithdrawn",
    options: EventOptions,
    cb: Callback<MIAWithdrawn>
  ): void;
}
