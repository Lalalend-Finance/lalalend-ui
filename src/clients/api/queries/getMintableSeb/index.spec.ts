import BigNumber from 'bignumber.js';

import vaiUnitrollerResponses from '__mocks__/contracts/sebUnitroller';
import fakeAddress from '__mocks__/models/address';
import { SebUnitroller } from 'types/contracts';

import getMintableSeb from '.';

describe('api/queries/getMintableSeb', () => {
  test('throws an error when request fails', async () => {
    const fakeContract = {
      methods: {
        getMintableVAI: () => ({
          call: async () => {
            throw new Error('Fake error message');
          },
        }),
      },
    } as unknown as SebUnitroller;

    try {
      await getMintableSeb({
        vaiControllerContract: fakeContract,
        accountAddress: fakeAddress,
      });

      throw new Error('getMintableSeb should have thrown an error but did not');
    } catch (error) {
      expect(error).toMatchInlineSnapshot('[Error: Fake error message]');
    }
  });

  test('returns the mintable VAI in the correct format on success', async () => {
    const callMock = jest.fn(async () => vaiUnitrollerResponses.getMintableVAI);
    const getMintableVAIMock = jest.fn(() => ({
      call: callMock,
    }));

    const fakeContract = {
      methods: {
        getMintableVAI: getMintableVAIMock,
      },
    } as unknown as SebUnitroller;

    const response = await getMintableSeb({
      vaiControllerContract: fakeContract,
      accountAddress: fakeAddress,
    });

    expect(callMock).toHaveBeenCalledTimes(1);
    expect(getMintableVAIMock).toHaveBeenCalledTimes(1);
    expect(response).toMatchSnapshot();
    expect(response.mintableSebWei instanceof BigNumber).toBeTruthy();
  });
});
