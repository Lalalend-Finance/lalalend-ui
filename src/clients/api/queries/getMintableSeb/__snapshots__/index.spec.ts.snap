// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`api/queries/getMintableSeb returns the mintable VAI in the correct format on success 1`] = `
Object {
  "mintableSebWei": "40000000000000000000",
}
`;
